src/main/java/com/openshift/internal/restclient/model/volume/VolumeSource.java

```diff
@@ -39,4 +39,35 @@ public String getName() {
 		return asString(getNode(), getPropertyKeys(), NAME);
 	}
 
+    @Override
+    public void setName(String name) {
+        set(getNode(), getPropertyKeys(), NAME, name);
+    }
+
+    @Override
+    public String toJSONString() {
+        if (!get(getNode(), getPropertyKeys(), NAME).isDefined()) {
+            throw new IllegalArgumentException("Name of volume source is missing");
+        }
+        return toJson(true);
+    }
+
+    public static IVolumeSource create(ModelNode node) {
+        if (node.has(VolumeType.EMPTY_DIR)) {
+            return new EmptyDirVolumeSource(node);
+        } else if (node.has(VolumeType.SECRET)) {
+            return new SecretVolumeSource(node);
+        } else if (node.has(VolumeType.PERSISTENT_VOLUME_CLAIM)) {
+            return new PersistentVolumeClaimVolumeSource(node);
+        } else {
+            return new UnsupportedVolumeSource(node);
```

Comment:
We should probably return an anonymous inner class implementation here instead of 'Unsupported' since we are returning something.  Unsupported makes it sound like we are unable to do anything without though its really some generic representation of a volume source since we have no specifics at this time.